### PR TITLE
Add media folder to perimeter

### PR DESCRIPTION
**Pull request by veolys-nanne**
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/2094
https://github.com/open-orchestra/open-orchestra-media-bundle/pull/247
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/343

### CODE REVIEW

File: MediaAdminBundle/GeneratePerimeter/Strategy/FolderGeneratePerimeterStrategy.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    protected $folderRepository;
+
+    /**
+     * @param FolderRepositoryInterface $folderRepository
+     * @param CurrentSiteIdInterface    $contextManager
+     */
+    public function __construct(
+        FolderRepositoryInterface $folderRepository,
+        CurrentSiteIdInterface $contextManager
+
```
Comment (ngilain) [inline MediaAdminBundle/GeneratePerimeter/Strategy/FolderGeneratePerimeterStrategy.php]: empty line to remove

File: MediaAdminBundle/GeneratePerimeter/Strategy/FolderGeneratePerimeterStrategy.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    /**
+     * Generate perimeter
+     *
+     * @return array
+     */
+    public function generatePerimeter()
+    {
+        $treeFolders = $this->folderRepository->findFolderTree($this->contextManager->getCurrentSiteId());
+
+        return $this->generateTreePerimeter(is_array($treeFolders) ? $treeFolders : array());
```
Comment (ngilain) [inline MediaAdminBundle/GeneratePerimeter/Strategy/FolderGeneratePerimeterStrategy.php]: folderRepository->findFolderTree always return an array so this test is useless

File: MediaAdminBundle/GeneratePerimeter/Strategy/FolderGeneratePerimeterStrategy.php [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    /**
+     * get perimeter configuration
+     *
+     * @return array
+     */
+    public function getPerimeterConfiguration()
+    {
+        $treeFolders = $this->folderRepository->findFolderTree($this->contextManager->getCurrentSiteId());
+
+        return $this->getTreePerimeterConfiguration(is_array($treeFolders) ? $treeFolders : array());
```
Comment (ngilain) [inline MediaAdminBundle/GeneratePerimeter/Strategy/FolderGeneratePerimeterStrategy.php]: same here

**Comment by veolys-nanne** [comment]
update done @ngilain

**Comment by ngilain** [comment]
:+1:

**Pull request merged**